### ISSUE TITLE
use-full-screen - ref returned is not memoized

### ISSUE THREAD

**Issue by ChaseMalik** [mantinedev/mantine#784]
### What package has an issue

@mantine/hooks

### Describe the bug

The ref function returned by useFullscreen is not wrapped in useCallback.

As a result, when passed to a dependency array (e.g. useMergedRef), it is executed on every render.

This seems like the change was introduced as part of 78a262e8d33604b866baec43acadf10807ea7c2f

### In which browser did the problem occur

All

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Wrap the ref function in a useCallback, as it was in the previous commit.

**Comment by rtivital** [comment]
Why would you need to pass this ref to a dependencies array?

**Comment by ChaseMalik** [comment]
I'm using useMergedRef, which passes it in the dependencies array.

I'm not sure exactly what the interaction is, but I'm using https://github.com/thebuilder/react-intersection-observer

And when I combine the two types of refs, I get infinite renders. Individually they both work fine though.

```
const ref = useRef();
const { ref: viewRef } = useInView();
const { ref: fullscreenRef } = useFullscreen();

// works
const mergedRef1 = useMergedRef(ref, viewRef);
// also works
const mergedRef2 = useMergedRef(ref, fullscreenRef);
// broken
const mergedRef3 = useMergedRef(ref, viewRef, fullscreenRef);
```

I understand this is partially related to the interaction with a different library, but wrapping fullscreenRef in a useCallback fixes it, and it is how it used to be in this code.

**Comment by rtivital** [comment]
Okay, thanks, I'll see what we can do about that

**Comment by rtivital** [comment]
Fixed in 3.6.7, ref is now memoized with useCallback
